src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/ReflectionEmitCachingMemberAccessor.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if NETFRAMEWORK || NETCOREAPP
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+
+namespace System.Text.Json.Serialization.Metadata
+{
+    internal sealed partial class ReflectionEmitCachingMemberAccessor : MemberAccessor
+    {
+        private static readonly ReflectionEmitMemberAccessor s_sourceAccessor = new();
+        private static readonly Cache<(string id, Type declaringType, MemberInfo? member)> s_cache =
+            new(slidingExpiration: TimeSpan.FromSeconds(5), evictionTriggerInterval: 200);
+
+        public override Action<TCollection, object?> CreateAddMethodDelegate<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)] TCollection>()
+            => s_cache.GetOrAdd((nameof(CreateAddMethodDelegate), typeof(TCollection), null), static _ => s_sourceAccessor.CreateAddMethodDelegate<TCollection>());
```

Comment:
@eerhardt this line is failing with the following trim analysis error:
```
Trim analysis error IL2091: System.Text.Json.Serialization.Metadata.ReflectionEmitCachingMemberAccessor.<>c__3<TCollection>.<CreateAddMethodDelegate>b__3_0(ValueTuple<String,Type,MemberInfo>): 'TCollection' generic argument does not satisfy 'DynamicallyAccessedMemberTypes.PublicMethods' in 'System.Text.Json.Serialization.Metadata.ReflectionEmitMemberAccessor.CreateAddMethodDelegate<TCollection>()'. The generic parameter 'TCollection' of 'System.Text.Json.Serialization.Metadata.ReflectionEmitCachingMemberAccessor.<>c__3<TCollection>' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to.
```
I suspect it's because the `TCollection` parameter within the lambda parameter is lifted to a generic parameter in the compiler generated method, which doesn't have the necessary attribute annotation. I'm not sure how I could make this error go away without substantially refactoring the code. Any suggestions?

Reply:
Nevermind, TIL I can attach attributes to lambda literals.

Reply:
You may also be hitting the issue described here: https://github.com/dotnet/roslyn/issues/46646

Reply:
Looks like the one, thanks. I should link to that in the justification string.